src/Saas.Permissions/Saas.Permissions.Api/Services/CertificateValidationService.cs

```diff
@@ -0,0 +1,24 @@
+﻿using Microsoft.Extensions.Configuration;
+using Saas.Permissions.Api.Interfaces;
+using System.Security.Cryptography.X509Certificates;
+namespace Saas.Permissions.Api.Services;
+
+public class CertificateValidationService : ICertificateValidationService
+{
+    private readonly IConfiguration _config;
+    public CertificateValidationService(IConfiguration config)
```

Comment:
This is generally not a good idea, since it adds a dependency to IConfiguration, where it really doesn't need IConfiguration it needs a Thumbprint.  More typing, but this is better: https://docs.microsoft.com/aspnet/core/fundamentals/configuration/options?view=aspnetcore-6.0

Reply:
Agreed, you map to an object and inject that. Options pattern is great.